Docs/setup ci

huhn511 opened the pull request:
Deploys mdbook and rust docs to github pages with a github action.

huhn511 committed:
fix structure

huhn511 committed:
add gh action

huhn511 committed:
fix readme

huhn511 committed:
push on develop changes and add cname

rajivshah3 commented on .github/workflows/deploy_docs_to_gh-pages.yml:
```diff
@@ -0,0 +1,24 @@
+      - develop
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
```
```suggestion
    runs-on: ubuntu-latest
```

huhn511 committed:
Update .github/workflows/deploy_docs_to_gh-pages.yml

Hmm, could this break the actions if there not compatible with ubuntu-20 for example? Or does it not matter?

Co-authored-by: Rajiv Shah <rajivshah1@icloud.com>

huhn511 committed:
move specs out of docs

huhn511 committed:
Merge branch 'docs/setup-ci' of https://github.com/huhn511/wallet.rs into docs/setup-ci

huhn511 committed:
fix merge conflict

lucasfernog committed:
Merge remote-tracking branch 'upstream/develop' into docs/setup-ci

huhn511 committed:
add rust docs to gh-pages

rajivshah3 commented on .github/workflows/deploy_docs_to_gh-pages.yml:
```diff
@@ -0,0 +1,27 @@
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup mdBook
```
```suggestion
      - name: Install rust-docs with nightly toolchain
        uses: actions-rs/toolchain@v1
        with:
            profile: minimal
            toolchain: nightly
            override: true
            components: rust-docs
          
      - name: Setup mdBook
```

I think this needs to be run because the default toolchain is `stable`

huhn511 committed:
Update .github/workflows/deploy_docs_to_gh-pages.yml

Co-authored-by: Rajiv Shah <rajivshah1@icloud.com>

huhn511 committed:
Merge remote-tracking branch 'upstream/develop' into docs/setup-ci

lucasfernog committed:
Merge remote-tracking branch 'upstream/develop' into docs/setup-ci
